spyder/plugins/plots/plugin.py

```diff
@@ -155,3 +186,415 @@ def apply_plugin_settings(self, options):
     def on_first_registration(self):
         """Action to be performed on first plugin registration"""
         self.tabify(self.main.variableexplorer)
+
+
+##############################################################################
+#
+# New API!
+#
+##############################################################################
+class PlotsWidget(SpyderWidget):
+    """
+    """
+    DEFAULT_OPTIONS = {
+        'auto_fit_plotting': True,
+        'mute_inline_plotting': True,
+        'show_plot_outline': True,
+        'save_dir': getcwd_or_home()
+    }
+
+    # Signals
+    sig_option_changed = Signal(str, object)
+
+    def __init__(self, parent, options=None):
+        super().__init__(parent=self, options=options)
+
+        # Widgets
+        self._stack = PlotsStackedWidget(parent=self)
+        self._shellwidgets = {}
+        self.zoom_disp = QSpinBox(self)
+
+        # Widget setup
+        self.zoom_disp.setAlignment(Qt.AlignCenter)
+        self.zoom_disp.setButtonSymbols(QSpinBox.NoButtons)
+        self.zoom_disp.setReadOnly(True)
+        self.zoom_disp.setSuffix(' %')
+        self.zoom_disp.setRange(0, 9999)
+        self.zoom_disp.setValue(100)
+
+        # Layout
+        layout = QHBoxLayout()
+        layout.addWidget(self._stack)
+        self.setLayout(layout)
+        self._stack.layout().setContentsMargins(0, 0, 0, 0)
+        self._stack.layout().setSpacing(0)
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.setSpacing(0)
+
+        # Signals
+        self._stack.sig_figure_menu_requested.connect(self.show_figure_menu)
+        self._stack.sig_thumbnail_menu_requested.connect(
+            self.show_thumbnail_menu)
+        self._stack.sig_zoom_changed.connect(self.zoom_disp.setValue)
+
+    # --- SpyderWidget API
+    # ------------------------------------------------------------------------
+    def get_title(self):
+        return _('Plots')
+
+    def get_focus_widget(self):
+        widget = self.current_widget()
+        if widget and widget.thumbnails_sb.current_thumbnail is not None:
+            if widget.figviewer.figcanvas.fig:
+                widget = widget.thumbnails_sb.scrollarea
+
+        return widget
+
+    def setup(self, options):
+        # Menu actions
+        mute_action = self.create_action(
```

Comment:
All these actions here @jitseniesen are exposed as shortcuts without having to register a shortcut manually.